tests/fuzz/fuzz-bech32.c

```diff
@@ -0,0 +1,49 @@
+#include <stdint.h>
+#include <string.h>
+#include <tests/fuzz/libfuzz.h>
+
+#include <common/bech32.h>
+
+void init(int *argc, char ***argv)
+{
+}
+
+void run(const uint8_t *data, size_t size)
+{
+	const char hrp_inv[5] = "lnbc\0", hrp_addr[3] = "bc\0";
+	char *bech32_str, *hrp_out, *addr;
+	uint8_t *data_out;
+	size_t data_out_len;
+	int wit_version;
+
+	/* Buffer size is defined in each function's doc comment. */
+	bech32_str = malloc(size + strlen(hrp_inv) + 8);
```

Comment:
?

Reply:
We usually use `tal` in C-Lightning, so a random `malloc` is something to note.  Is there a justification for using `malloc` instead of `tal`?  E.g. "we inadvertently end up testing the `tal` code instead of whatever this test is testing if we use `tal`, so use `malloc` instead."

Reply:
I only used `tal` when required, here this interfaces with a header that has no concept of `tal` (even gives the size to allocate in headers)